src/python/pants/backend/python/dependency_inference/import_parser.py

```diff
@@ -56,15 +57,16 @@ def parse_file(*, filename: str, content: str) -> Optional[Tuple]:
             return None
 
 
-def find_python_imports(*, filename: str, content: str, module_name: str) -> ParsedPythonImports:
+def find_python_imports(*, filename: str, content: str) -> ParsedPythonImports:
+    package = ".".join(PurePath(filename).parts[0:-1])
```

Comment:
You join here just to split in `_BaseAstVisitor` - how about just pass the parts - maybe in a tuple newtype if having the name/type `Package` will provide a bit more clarity.

Reply:
Not a huge deal either way how this is factored. This code is all being rewritten atm to run via a `python -c` subprocess.